### DISCUSSION TITLE
Question Regarding OpenSSL Upgrades and Application Compatibility

### DISCUSSION THREAD

**Discussion by adeshmrane** [openssl/openssl#26214] [Q&A] [answered]
We have some applications (including Apache, Nginx, PHP, and curl) that were manually compiled against **_OpenSSL 3.0.7_**. We are considering upgrading OpenSSL to version **_3.0.15_** and would like to understand the potential impact on these applications.

Specifically, we are concerned about whether the compiled binaries will continue to function correctly after the OpenSSL upgrade, or if recompilation will be required. Could you please advise on the compatibility between these versions and any potential issues we might encounter?

We also have systems using OpenSSL **_1.1.1s_** and are planning to upgrade to **_1.1.1w._** We have the same concerns regarding compatibility and would appreciate your insights on this upgrade path as well.

Thank you for your time and assistance. We look forward to your response.

**Comment by t8m** [comment, marked as answer]
With 3.x versions, the applications compiled against the older minor versions will work fine with newer versions.

I.e., you can build with 3.0.7 and run with 3.0.15, or even 3.4.0.

With 1.x versions the same constraints hold for patch versions. I.e., An application built with 1.1.0c should be able run with 1.1.1x.

**Reply by adeshmrane** [reply]
Thank you for confirming this. We appreciate your prompt and helpful response.
